components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -947,35 +975,92 @@ private boolean isUsernameCaseSensitive(String tenantQualifiedUsername) {
      * @return
      * @throws IdentityOAuth2Exception
      */
+    /**
+     * @deprecated use {@link #getOIDCIdTokenAudiences(String, String )} instead.
+     */
+    @Deprecated
     public List<String> getOIDCAudiences(String tenantDomain, String consumerKey) throws IdentityOAuth2Exception {
 
-        List<String> audiences = new ArrayList<>();
+        return this.getOIDCIdTokenAudiences(tenantDomain, consumerKey);
+
+    }
+
+    /**
+     * Retrieves OIDC ID Token audience values configured for an oauth consumer app.
+     *
+     * @param tenantDomain application tenant domain
+     * @param consumerKey  client ID
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public List<String> getOIDCIdTokenAudiences(String tenantDomain, String consumerKey)
```

Comment:
The getters and setters can be refactored by passing another parameter (boolean).

Reply:
resolved with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1902/commits/969e3978becce615f911d41d1b72521ce2bde2b2